modules/standard/FileSystem.chpl

```diff
@@ -861,6 +869,10 @@ proc isLink(out error:syserr, name: string): bool {
   extern proc chpl_fs_is_link(ref result:c_int, name: c_string): syserr;
 
   var ret:c_int;
+  var doesExist = exists(error, name);
```

Comment:
Thinking about this a little bit more, we probably shouldn't use `exists` for any of these as we don't want to indicate an error. We should probably check the errors of `stat` and `lstat` (`ENOENT` and `ENOTDIR`, maybe `EACCES`, `ENAMETOOLONG` and others) and then return the appropriate value.